Configuration/src/main/java/io/deephaven/configuration/Configuration.java

```diff
@@ -27,7 +29,8 @@ public class Configuration extends PropertyFile {
     private static final Logger log = LoggerFactory.getLogger(Configuration.class);
 
     private static Configuration DEFAULT;
-    private static final Map<String, Configuration> NAMED_CONFIGURATIONS = new HashMap<>();
+    private static final Map<String, Configuration> NAMED_CONFIGURATIONS = new ConcurrentHashMap<>();
+    private final Supplier<ConfigurationContext> contextSupplier;
```

Comment:
Do we think it is worth having a supplier instead of an actual object here; if we think we would create it again and again I can see wanting something fresh.  But I'm not sure if the level of indirection is buys us anything or just adds a little complexity.

I'm fine either way, since I think this will accomplish our goals.

Reply:
I chose a supplier because otherwise the semantics of reloadConfiguraiton() changes.  In the current impl it always creates a new instance.  I wanted to keep that the same.